packages/backend/src/models/JobModel/JobModel.ts

```diff
@@ -85,44 +84,59 @@ export class JobModel {
                 stepType,
                 stepError: step.step_error,
                 stepLabel,
+                startedAt: step.started_at,
             };
         });
     }
 
-    async upsertJobStatus(
+    async update(
         jobUuid: string,
-        projectUuid: string | undefined,
-        status: JobStatusType,
+        job: Partial<Pick<Job, 'jobStatus' | 'jobType' | 'jobResults'>>,
     ): Promise<void> {
-        Logger.debug(
-            `Updating job status ${jobUuid} for project ${projectUuid} with status ${status}`,
-        );
         await this.database(JobsTableName)
-            .insert({
-                project_uuid: projectUuid,
-                job_uuid: jobUuid,
-                job_status: status,
-                updated_at: new Date(),
+            .update({
+                ...(job.jobStatus !== undefined
+                    ? { job_status: job.jobStatus }
+                    : {}),
+                ...(job.jobType !== undefined ? { job_type: job.jobType } : {}),
+                ...(job.jobResults !== undefined
+                    ? { results: job.jobResults }
+                    : {}),
             })
-            .onConflict('job_uuid')
-            .merge();
+            .where('job_uuid', jobUuid);
     }
 
-    async createJobSteps(
-        jobUuid: string,
-        stepTypes: JobStepType[],
-    ): Promise<void> {
+    async create(job: CreateJob): Promise<Job> {
         await this.database.transaction(async (trx) => {
-            await stepTypes.reduce(async (previousPromise, step) => {
+            const [row] = await trx(JobsTableName)
+                .insert({
+                    project_uuid: job.projectUuid,
+                    job_uuid: job.jobUuid,
+                    job_type: job.jobType,
+                    job_status: job.jobStatus,
+                })
+                .returning('*');
+            const { job_uuid: jobUuid } = row;
+            await job.steps.reduce(async (previousPromise, step) => {
                 await previousPromise;
                 return trx(JobStepsTableName).insert({
                     job_uuid: jobUuid,
                     step_status: JobStepStatusType.PENDING,
-                    step_type: step,
-                    step_error: undefined,
+                    step_type: step.stepType,
                 });
             }, Promise.resolve());
         });
+        return this.get(job.jobUuid);
+    }
+
+    async startJobStep(jobUuid: string, stepType: JobStepType): Promise<void> {
+        await this.database(JobStepsTableName)
+            .update({
+                step_status: JobStepStatusType.RUNNING,
+                started_at: new Date(),
+            })
+            .where('job_uuid', jobUuid)
+            .andWhere('step_type', stepType);
```

Comment:
do we need to add an extra filter `.andWhere('step_status', PENDING)` to make sure we are not running `skipped` steps

Reply:
Following your previous comment, I think we should try and keep this business logic out of the model